### PR TITLE
SNOW-828572 Set UseProxy on HttpClientHandler

### PR DESCRIPTION
**Pull request by sfc-gh-knozderko**
It is a bug fix that resolves the issue: https://github.com/snowflakedb/snowflake-connector-net/issues/653
The fix is about setting properly UseProxy attribute of HttpClientConfig to indicate whether a proxy is being used.

### CODE REVIEW

File: Snowflake.Data.Tests/HttpUtilTest.cs [hunk 1]
```diff
@@ -36,5 +38,51 @@ public async Task TestIsRetryableHTTPCode(HttpStatusCode statusCode, bool forceR
 
             Assert.AreEqual(expectedIsRetryable, actualIsRetryable);
         }
+
+        [Test]
+        public void ShouldCreateHttpClientHandlerWithProxy()
+        {
+            // given
```
Comment (sfc-gh-dprzybysz) [inline Snowflake.Data.Tests/HttpUtilTest.cs:45]: is it a new practice to write comments give/when/then

Comment (sfc-gh-knozderko) [reply Snowflake.Data.Tests/HttpUtilTest.cs:45]: It is quite a new practice (previously I had added this convention in a few tests).
But even though previously it was not followed why not to introduce a good practice?

File: Snowflake.Data.Tests/HttpUtilTest.cs [hunk 2]
```diff
@@ -36,5 +38,51 @@ public async Task TestIsRetryableHTTPCode(HttpStatusCode statusCode, bool forceR
...
+            var config = new HttpClientConfig(
+                true,
+                "snowflake.com",
+                "123",
+                "Bartek",
```
Comment (sfc-gh-dstempniak) [inline Snowflake.Data.Tests/HttpUtilTest.cs:50]: maybe use "test" instead of Bartek :)

**Review by sfc-gh-igarish — APPROVED** [review]
LGTM

File: Snowflake.Data.Tests/HttpUtilTest.cs (hunk 2 above)
Comment (sfc-gh-knozderko) [reply Snowflake.Data.Tests/HttpUtilTest.cs]: changed

**Review by sfc-gh-mhofman — APPROVED** [review]

File: Snowflake.Data/Core/HttpUtil.cs [hunk 3]
```diff
@@ -127,7 +128,8 @@ private HttpMessageHandler setupCustomHttpHandler(HttpClientConfig config)
                 httpHandler = new HttpClientHandler()
                 {
                     AutomaticDecompression = DecompressionMethods.GZip | DecompressionMethods.Deflate,
-                    UseCookies = false // Disable cookies
+                    UseCookies = false, // Disable cookies
```
Comment (sfc-gh-mhofman) [inline Snowflake.Data/Core/HttpUtil.cs:131]: I would remove this comment

File: Snowflake.Data/Core/HttpUtil.cs [hunk 4]
```diff
@@ -117,7 +117,8 @@ private HttpMessageHandler setupCustomHttpHandler(HttpClientConfig config)
                     // Enforce tls v1.2
                     SslProtocols = SslProtocols.Tls12,
                     AutomaticDecompression = DecompressionMethods.GZip | DecompressionMethods.Deflate,
-                    UseCookies = false // Disable cookies
+                    UseCookies = false, // Disable cookies
```
Comment (sfc-gh-mhofman) [inline Snowflake.Data/Core/HttpUtil.cs:120]: I would remove this comment

**Pull request merged**